e2e_test/js-tests/test_viem_smoketest.mjs

```diff
@@ -0,0 +1,170 @@
+	const transaction = await publicClient.getTransaction({ hash: txHash });
+	assert.equal(transaction.type, type, "transaction type does not match");
+	assert.equal(receipt.status, "success", "receipt status 'failure'");
+}
+
+// sendTypedTransaction sends a transaction with the given type and an optional
+// feeCurrency.
+async function sendTypedTransaction(type, feeCurrency){
+		 return await walletClient.sendTransaction({
+			to: "0x00000000000000000000000000000000DeaDBeef",
+			value: 1,
+			type: type,
+			feeCurrency: feeCurrency,
+		});
+}
+
+// sendTypedCreateTransaction sends a create transaction with the given type
+// and an optional feeCurrency.
+async function sendTypedCreateTransaction(type, feeCurrency){
+		 return await walletClient.deployContract({
+			type: type,
+			feeCurrency: feeCurrency,
+			bytecode: testContractJSON.bytecode.object,
+			abi: testContractJSON.abi,
+			 // The constructor args for the test contract at ../debug-fee-currency/DebugFeeCurrency.sol
+			args:[1n, true, true, true],
+		});
+}
+
+// verifyTypedTransactions is a helper function that submits a send and create
+// transaction of the given type with optional feeCurrency and checks the
+// results.
+async function verifyTypedTransactions(type, feeCurrency){
+		const send = await sendTypedTransaction(type, feeCurrency);
+		await check(send, type);
+		const create = await sendTypedCreateTransaction(type, feeCurrency);
+		await check(create, type);
+}
+
+describe("viem smoke test", () => {
+	it("send legacy tx", async () => {
+		const type = "legacy";
+		await verifyTypedTransactions(type);
+	}).timeout(30_000);
+
+	it("send eip2930 tx", async () => {
+		const type = "eip2930";
+		await verifyTypedTransactions(type);
+	}).timeout(10_000);
+
+	it("send eip1559 tx", async () => {
+		const type = "eip1559";
+		await verifyTypedTransactions(type);
+	}).timeout(10_000);
+
+	it("send cip64 tx", async () => {
+		const type = "cip64";
+		await verifyTypedTransactions(type, process.env.FEE_CURRENCY);
+	}).timeout(10_000);
+
+	// This test is failing because the produced transaction is of type cip64.
+	// I guess this is a problem with the viem internals.
+	it.skip("cip42 not supported", async () => {
+		const type = "cip42";
+		await verifyTypedTransactions(type);
+	}).timeout(10_000);
+
+	it.only("legacy tx with fee currency not supported", async () => {
+		const type = "legacy";
+		try {
+			const hash = await sendTypedTransaction(type,process.env.FEE_CURRENCY);
+
+			// When using the devnet an exception is thrown from
+			// sendTypedTransaction, on alfajores the transaction is submitted so we
+			// get a hash but it's not included. So we validate here that the
+			// transaction was not included.
+			let blockNumber = await publicClient.getBlockNumber();
+			const oldBlockNumber = blockNumber;
+			while(blockNumber == oldBlockNumber){
+				blockNumber = await publicClient.getBlockNumber();
+			}
```

Comment:
A small sleep (e.g. 50ms) would be prevent people with low latency connections (or a script in the same datacenter) to send a large amount of requests while waiting. `getBlockNumber` should be fast enough for that to not matter too much, but let's avoid being spammy on principle.